master/internal/db/postgres_experiments_intg_test.go

```diff
@@ -379,6 +379,119 @@ func TestMetricNames(t *testing.T) {
 	require.Equal(t, []string{"b", "c", "f"}, actualNames[model.MetricGroup("golabi")])
 }
 
+func TestExperimentByIDs(t *testing.T) {
+	ctx := context.Background()
+
+	require.NoError(t, etc.SetRootPath(RootFromDB))
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, MigrationsFromDB)
+	user := RequireMockUser(t, db)
+
+	externalID := uuid.New().String()
+	exp := RequireMockExperimentParams(t, db, user, MockExperimentParams{
+		ExternalExperimentID: &externalID,
+	})
+	trial, task := RequireMockTrial(t, db, exp)
+
+	for _, c := range []struct {
+		funcName string
+		f        func(IsNotFoundCase bool) (*model.Experiment, error)
+	}{
+		{"ExperimentByID", func(isNotFoundCase bool) (*model.Experiment, error) {
+			if isNotFoundCase {
+				return ExperimentByID(ctx, -1)
+			}
+
+			return ExperimentByID(ctx, exp.ID)
+		}},
+		{"ExperimentByTrialID", func(isNotFoundCase bool) (*model.Experiment, error) {
+			if isNotFoundCase {
+				return ExperimentByTrialID(ctx, -1)
+			}
+
+			return ExperimentByTrialID(ctx, trial.ID)
+		}},
+		{"ExperimentByTaskID", func(isNotFoundCase bool) (*model.Experiment, error) {
+			if isNotFoundCase {
+				return ExperimentByTaskID(ctx, model.TaskID(uuid.New().String()))
+			}
+
+			return ExperimentByTaskID(ctx, task.TaskID)
+		}},
+		{"ExperimentByExternalIDTx", func(isNotFoundCase bool) (*model.Experiment, error) {
+			if isNotFoundCase {
+				return ExperimentByExternalIDTx(ctx, Bun(), uuid.New().String())
+			}
+
+			return ExperimentByExternalIDTx(ctx, Bun(), externalID)
+		}},
+	} {
+		t.Run(c.funcName+"-notFound", func(t *testing.T) {
+			_, err := c.f(true)
+			require.ErrorIs(t, err, ErrNotFound)
+		})
+
+		t.Run(c.funcName, func(t *testing.T) {
+			e, err := c.f(false)
+			require.NoError(t, err)
+
+			require.Equal(t, e, exp)
```

Comment:
nit (non-blocking): Do we need a space in between these two require statements?

Reply:
I personally like the space just because it highlights the error check is with ```e, err := c.f(false)```. Then we can do something with the value after the error check is done.

I don't care too strongly either way though